packages/@icon-magic/icon-models/src/asset.ts

```diff
@@ -35,43 +36,39 @@ export class Asset {
     // the file itself
     this.name = config.name ? config.name : path.parse(config.path).name;
 
-    // set the path (this automatically calls the setter defined on path)
+    // set the path
     this.path = config.path;
 
     // set the contents only if it is passed in the config
     if (config.contents) {
       this.contents = config.contents;
     }
+    this.assetConfig =  {
+      name: this.name,
+      path: this.path
+    };
     this.debug(`Asset ${this.name} created in ${this.iconPath}`);
   }
 
   /**
-   * If an absolute path is passed in, store the relative path relative to the
-   * iconPath
+   * @returns The path of an asset which is always relative to the iconpath
    */
-  set path(filePath: string) {
-    if (path.isAbsolute(filePath)) {
-      this.unresolvedPath = path.relative(this.iconPath, filePath);
-    } else {
-      this.unresolvedPath = filePath;
-    }
+  getPath() {
+    return path.resolve(this.iconPath, this.path);
   }
 
   /**
-   * The path of an asset is always relative to the iconpath
+   * Sets the path property to a path passed in
    */
-  get path() {
-    return path.resolve(this.iconPath, this.unresolvedPath);
+  setPath(path: string) {
```

Comment:
Should we turn on the linting rule that requires an explicit `void` declaration for functions that aren't expected to return anything?

Reply:
added `"typedef": [true, "call-signature"]` (https://palantir.github.io/tslint/rules/typedef/) but not sure it's working...

Reply:
Added the rule, doesn't look like it does anything right now, will look again in another PR.

Reply:
Update, it works, will fix errors in later PR :)